<a href="https://colab.research.google.com/github/IsaiGowthami/Excelr_Assignments/blob/main/chisquaretest.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import pandas as pd
from scipy.stats import chi2_contingency

In [ ]:
data={'smart Thermostat':[50,80,60,30,20],
      'smart light':[70,100,90,50,50]}
satisfaction=['Very Satisfied','Satisfied','Neutral','Unsatisfied','Very Unsatisfied']
df=pd.DataFrame(data, index=satisfaction)
chi2_stat,p_value,dof,expected=chi2_contingency(df)
alpha=0.05
if p_value<alpha:
    print('Reject the null hypothesis:There is a significant association between device type and satisfaction level.')
else:
    print('Fail to reject the null hypothesis: There is no significant association between device type and satisfaction level.')

Fail to reject the null hypothesis: There is no significant association between device type and satisfaction level.


In [ ]:
#just showing the chi square statistic, crtical values, degrees of freedom, p value for better understanding.
print('Chi-Square Statistic:',chi2_stat)
print('P-Value:',p_value)
print('Degrees of Freedom:',dof)
print('Expected Frequencies:',expected)

Chi-Square Statistic: 5.638227513227513
P-Value: 0.22784371130697179
Degrees of Freedom: 4
Expected Frequencies: [[ 48.  72.]
 [ 72. 108.]
 [ 60.  90.]
 [ 32.  48.]
 [ 28.  42.]]


Conclusion:

Chi-Square Statistic: 5.638

Critical Value: 9.488

Degrees of Freedom: 4

P-Value: 0.228

Since the Chi-Square statistic (5.638) is less than the critical value (9.488), we fail to reject the null hypothesis. This means there is no significant association between the type of smart home device purchased and the customer satisfaction level at the 0.05 significance level.

[Hypothesis Statement:

The null hypothesis (H₀) is that there is no association between device type and customer satisfaction level.

The alternative hypothesis (H₁) is that there is an association between the two.]

here instead of using the manual formulae to calculate all the values I have directly imported the necessary libraries which have internally performed calculations and given us the right outputs/conclusions.
